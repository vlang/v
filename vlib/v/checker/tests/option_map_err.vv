struct OptionalMap {
	opmap ?map[string]f64
}

fn main() {
	op := OptionalMap{
		opmap: {
			'1': 0.0
		}
	}
	assert op.opmap or {
		{
			'1': 0.0
		}
	}['1'] == 0.0

	op2 := OptionalMap{
		opmap: {
			'1': 1.0
		}
	}
	assert op2.opmap['1'] == 1.0
}
