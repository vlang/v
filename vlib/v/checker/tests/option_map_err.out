vlib/v/checker/tests/option_map_err.vv:22:13: error: type `?map[string]f64` is an Option, it must be unwrapped first; use `op2.opmap?` to do it
   20 |         }
   21 |     }
   22 |     assert op2.opmap['1'] == 1.0
      |                ~~~~~
   23 | }
vlib/v/checker/tests/option_map_err.vv:22:13: error: field `opmap` is an Option, so it should have either an `or {}` block, or `?` at the end
   20 |         }
   21 |     }
   22 |     assert op2.opmap['1'] == 1.0
      |                ~~~~~
   23 | }
